snuba/cli/bootstrap.py

```diff
@@ -57,22 +57,28 @@ def bootstrap(
                     raise
                 time.sleep(1)
 
-        topics = []
+        topics = dict()
         for name in DATASET_NAMES:
             dataset = get_dataset(name)
             table_writer = dataset.get_table_writer()
             if table_writer:
                 stream_loader = table_writer.get_stream_loader()
                 for topic_spec in stream_loader.get_all_topic_specs():
-                    topics.append(
-                        NewTopic(
-                            topic_spec.topic_name,
-                            num_partitions=topic_spec.partitions_number,
-                            replication_factor=topic_spec.replication_factor,
-                        )
+                    if topic_spec.topic_name in topics:
+                        continue
+                    logger.debug(
+                        "Adding topic %s to creation list", topic_spec.topic_name
+                    )
+                    topics[topic_spec.topic_name] = NewTopic(
+                        topic_spec.topic_name,
+                        num_partitions=topic_spec.partitions_number,
+                        replication_factor=topic_spec.replication_factor,
                     )
 
-        for topic, future in client.create_topics(topics).items():
+        logger.debug("Initiating topic creation")
+        for topic, future in client.create_topics(
+            list(topics.values()), operation_timeout=1
```

Comment:
I don't have any specific reason to think this, but I wouldn't be surprised if this timeout is too low.

Reply:
Actually, maybe not. I was just testing something else out with this `operation_timeout` argument locally — without a broker running, setting this to any value didn't prevent `future.result` from blocking indefinitely. `request_timeout` does work here when the broker can't be reached.

Reply:
>without a broker running, setting this to any value didn't prevent future.result from blocking indefinitely. request_timeout does work here when the broker can't be reached

I think the previous loop is enough to make sure we have a broker reachable but I'm okay removing that part and baking the retry/request timeout logic into this part (which means we'll have this part in a loop too).

If you think this is a good idea, I'll submit a follow-up patch.

Reply:
> I think the previous loop is enough to make sure we have a broker reachable

Oh right, good point. I think this is fine as-is.